cinema/views.py

```diff
@@ -1,13 +1,15 @@
 from datetime import datetime
 
 from django.db.models import F, Count
+from drf_spectacular.types import OpenApiTypes
+from drf_spectacular.utils import extend_schema, OpenApiParameter
 from rest_framework import viewsets, mixins, status
 from rest_framework.authentication import TokenAuthentication
 from rest_framework.decorators import action
 from rest_framework.pagination import PageNumberPagination
 from rest_framework.permissions import IsAuthenticated, IsAdminUser
 from rest_framework.response import Response
-from rest_framework.viewsets import GenericViewSet, ReadOnlyModelViewSet
+from rest_framework.viewsets import GenericViewSet
```

Comment:
1. You set DEFAULT_AUTHENTICATION_CLASSES in settings.py but your viewsets still override these auth class.
2. Do you need here IsAuthenticated permission if you set auth class?
3. After 1 and 2 are fixed, remove appropriate imports you don't use